### DISCUSSION TITLE
How does scalar index supports  integer, and float type fields

### DISCUSSION THREAD

**Discussion by king1302217** [milvus-io/milvus#34429] [Q&A and General discussion]
According to https://milvus.io/docs/scalar_index.md#Scalar-Index，scalar index is base on Inverted indexing。An inverted index has two main components: a term dictionary and an inverted list.  As far as I know，  inverted index only support varchar field。 So how does scalar index supports  integer, and float type fields？

**Comment by yhmo** [comment]
Milvus inverted index is based on [Tantivy](https://github.com/quickwit-oss/tantivy?tab=readme-ov-file#features). You can read articles about tantivy to learn more.

**Reply by xiaofan-luan** [reply]
usually, you can use stlsort index for numeric numbers. 
I guess tantivy simply reorder the data as inverted index

**Reply by king1302217** [reply]
@xiaofan-luan How can I set  stlsort index in milvus?

**Reply by xiaofan-luan** [reply]
https://milvus.io/docs/index-scalar-fields.md 

check this document
